cardano-api/gen/Gen/Cardano/Api/Typed.hs

```diff
@@ -772,11 +770,13 @@ genExecutionUnits :: Gen ExecutionUnits
 genExecutionUnits = ExecutionUnits <$> Gen.integral (Range.constant 0 1000)
                                    <*> Gen.integral (Range.constant 0 1000)
 
--- | 'Gen.maybe' but with condition if era is Alonzo-based
-ifAlonzoBased :: CardanoEra era -> Gen a -> Gen (Maybe a)
-ifAlonzoBased era gen
-  | isAlonzoBased era = Gen.maybe gen
-  | otherwise         = pure Nothing
+-- | Gen for Alonzo-specific parameters
+alonzoParam :: CardanoEra era -> Gen a -> Gen (Maybe a)
+alonzoParam era gen =
+  case era of
+    AlonzoEra             -> Just <$> gen
+    _ | isAlonzoBased era -> Gen.maybe gen
```

Comment:
Why this extra case with `isAlonzoBased`?  Why not pattern match on all the eras instead?

Reply:
I have rewritten this. Please take a look.